framework/src/problems/FEProblemBase.C

```diff
@@ -4315,12 +4315,14 @@ FEProblemBase::addUserObject(const std::string & user_object_name,
     // Account for displaced mesh use
     if (_displaced_problem && parameters.get<bool>("use_displaced_mesh"))
     {
-      if (euo || nuo || duo)
+      // Whether to re-init or not depends on the attributes of the base classes.
+      // For example, InterfaceUOBase has "_current_side_elem" and "_neighbor_elem"
+      // so it needs to reinit on displaced faces and neighbors
+      if (euo || nuo || duo || isuo || iuob)
         _reinit_displaced_elem = true;
-      else if (suo || duo)
-        // shouldn't we add isuo
+      if (suo || duo || isuo || iuob)
         _reinit_displaced_face = true;
-      else if (iuob || duo)
+      if (iuob || duo || isuo || iuob)
```

Comment:
I bet these variable names are Robert's work 😆 I don't like them because they're so short. Would you be willing to at least change `iuob` to `iuo`? The `b` suffix makes me first think of "boundary" instead of "base"